### PR TITLE
[syncd] Add workaround for port error status notification

### PR DESCRIPTION
**Pull request by kcudnik**
Adding new field in port status notification is breaking
change and can cause some issues when getting notification
from older SAI library, this will add workaround to handle
that case

### CODE REVIEW

**Comment by kcudnik** [comment]
I'm just creating this PR to see what i miss from build, i will later split this to 2 PR's as advancing SAI submodule and then commiting actual workaround

File: syncd/VendorSai.cpp [hunk 1]
```diff
@@ -1300,6 +1354,52 @@ sai_status_t VendorSai::bulkRemove(
             object_statuses);
 }
 
+sai_status_t VendorSai::bulkRemove(
```
Comment (prgeor) [inline syncd/VendorSai.cpp:1357]: @kcudnik I see this change is not relevant as per the PR title. Can this be kept outside this PR and added in another PR?

File: syncd/VendorSai.cpp [hunk 2]
```diff
@@ -998,6 +998,60 @@ sai_status_t VendorSai::bulkCreate(
             object_statuses);
 }
 
+sai_status_t VendorSai::bulkCreate(
```
Comment (prgeor) [inline syncd/VendorSai.cpp:1001]: @kcudnik this change be added as another PR as this is NOT relevant as per the PR title

File: syncd/VendorSai.cpp (hunk 1 above)
Comment (kcudnik) [reply syncd/VendorSai.cpp:1357]: read this comment: https://github.com/sonic-net/sonic-sairedis/pull/1430#issuecomment-2405155952, this is 1st pr: https://github.com/sonic-net/sonic-sairedis/pull/1431

File: syncd/VendorSai.cpp (hunk 2 above)
Comment (kcudnik) [reply syncd/VendorSai.cpp:1001]: read this comment: https://github.com/sonic-net/sonic-sairedis/pull/1430#issuecomment-2405155952, this is 1st pr: https://github.com/sonic-net/sonic-sairedis/pull/1431

**Comment by prgeor** [comment]
@kcudnik could you please check the build failure.

**Comment by kcudnik** [comment]
Yes, already fixed in latest pr #1431 but swss repo tests are failing because of sai backward compatibility break, I already fixed that and posted pr on swss repo pleased follow thread on #1431

**Comment by kcudnik** [comment]
SAI was updated in separate PR: https://github.com/sonic-net/sonic-sairedis/pull/1431

File: syncd/NotificationHandler.cpp [hunk 3]
```diff
@@ -93,7 +111,9 @@ void NotificationHandler::onPortStateChange(
 {
     SWSS_LOG_ENTER();
 
-    auto s = sai_serialize_port_oper_status_ntf(count, data);
+    auto ntfdata = Workaround::convertPortOperStatusNotification(count, data, m_apiVersion);
+
+    auto s = sai_serialize_port_oper_status_ntf((uint32_t)ntfdata.size(), ntfdata.data());
```
Comment (prgeor) [inline syncd/NotificationHandler.cpp:116]: @kcudnik I think you need to parse the new field `sai_port_error_status_t` in sai_serialize_port_oper_status_ntf() otherwise this new field will not be posted to redis DB and OA will not see this field in the notification

**Review by prgeor — APPROVED** [review]

**Comment by kcudnik** [comment]
@prgeor why there was force push to my branch from you?
